.github/workflows/ci.yaml

```diff
@@ -168,13 +177,44 @@ jobs:
       - name: Install Numaplane
         env:
           GOPATH: /home/runner/go
-        run:
-          KUBECONFIG=~/.kube/config VERSION=${{ github.sha }} DOCKER_PUSH=true STRATEGY=${{matrix.strategy}} make start
+        run: KUBECONFIG=~/.kube/config VERSION=${{ github.sha }} DOCKER_PUSH=true STRATEGY=${{matrix.strategy}} TEST_TYPE=e2e make start
+
+      - name: Wait for Numaplane to be ready
+        run: |
+          export POD_NAME=$(kubectl get pods -n numaplane-system -l app.kubernetes.io/name=controller-manager -o jsonpath="{.items[0].metadata.name}")
+          echo 'Waiting for the numaplane-controller-manager to be ready...'
+          kubectl get po -n numaplane-system
+          kubectl wait --for=condition=Ready --timeout=5m pod/$POD_NAME -n numaplane-system
 
       - name: Run e2e tests
         env:
           GOPATH: /home/runner/go
-        run: KUBECONFIG=~/.kube/config VERSION=${{ github.sha }} DOCKER_PUSH=true STRATEGY=${{matrix.strategy}} ENABLE_POD_LOGS=true make test-${{matrix.case}}-e2e
+        run: |
+          KUBECONFIG=~/.kube/config VERSION=${{ github.sha }} DOCKER_PUSH=true STRATEGY=${{matrix.strategy}} ENABLE_POD_LOGS=true make test-${{matrix.case}}-e2e
+
+      ## This is a temporary solution to kill the process twice, until we have fix for a stuck process after initial kill.
```

Comment:
that's interesting that using SIGTERM twice works. I guess SIGTERM allows the program to try to shut down gracefully, right?

If it doesn't work the first time, I don't know why it would work the second time. ?

I would think it would need to be SIGTERM followed by SIGKILL (Not that I want to hold up this PR from this but it's weird)